### ISSUE TITLE
fct_collapse applying collapsed factors in wrong order when group_other = TRUE

### ISSUE THREAD

**Issue by gtm19** [tidyverse/forcats#172]
I'm having an issue which is self explanatory from the title. Reprex below. As you can see, the classification produced by `fct_collapse` is incorrect.

As requested, a simpler set of examples is below.

## Example 1 - Collapsing _Character_ Variables
The `fct_collapse` function wrongly classifies _raspberry_ as a vegetable, rather than broccoli.
```r
library(forcats)
library(dplyr)
library(tibble)

df <-
  tibble(item = c("apple", "grape", "banana", "broccoli", "raspberry"))

df %>% 
  mutate(category = fct_collapse(item,
                                 fruit = c("apple", "grape", "banana", "raspberry"),
                                 vegetables = "broccoli", group_other = TRUE))
#> # A tibble: 5 x 2
#>   item      category  
#>   <chr>     <fct>     
#> 1 apple     fruit     
#> 2 grape     fruit     
#> 3 banana    fruit     
#> 4 broccoli  fruit     
#> 5 raspberry vegetables
```
<details>

## Example 2 - Collapsing _Factor_Variables
For clarity, and contrary to my previous explanation, this bug affects both character and factor variables (since `fct_collapse` converts the `.f` argument to a factor if it is a character variable):
```r
df <-
  tibble(item = factor(c("apple", "grape", "banana", "broccoli", "raspberry")))

df %>% 
  mutate(category = fct_collapse(item,
                                 fruit = c("apple", "grape", "banana", "raspberry"),
                                 vegetables = "broccoli", group_other = TRUE))
#> # A tibble: 5 x 2
#>   item      category  
#>   <fct>     <fct>     
#> 1 apple     fruit     
#> 2 grape     fruit     
#> 3 banana    fruit     
#> 4 broccoli  fruit     
#> 5 raspberry vegetables
```
## Example 3 - with `group_other` argument set to `FALSE`
Just to demonstrate that this bug does not occur when `group_other` is set to `FALSE`:
```r
df <-
  tibble(item = factor(c("apple", "grape", "banana", "broccoli", "raspberry")))

df %>% 
  mutate(category = fct_collapse(item,
                                 fruit = c("apple", "grape", "banana", "raspberry"),
                                 vegetables = "broccoli", group_other = FALSE))
#> # A tibble: 5 x 2
#>   item      category  
#>   <fct>     <fct>     
#> 1 apple     fruit     
#> 2 grape     fruit     
#> 3 banana    fruit     
#> 4 broccoli  vegetables
#> 5 raspberry fruit

```
## Session Info
```r
sessionInfo()
#> R version 3.5.2 (2018-12-20)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 17134)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=English_United Kingdom.1252 
#> [2] LC_CTYPE=English_United Kingdom.1252   
#> [3] LC_MONETARY=English_United Kingdom.1252
#> [4] LC_NUMERIC=C                           
#> [5] LC_TIME=English_United Kingdom.1252    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] bindrcpp_0.2.2     tibble_2.0.1       dplyr_0.7.8       
#> [4] forcats_0.4.0.9000
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.0       knitr_1.21       bindr_0.1.1      magrittr_1.5    
#>  [5] tidyselect_0.2.5 R6_2.3.0         rlang_0.3.1      fansi_0.4.0     
#>  [9] stringr_1.4.0    highr_0.7        tools_3.5.2      xfun_0.4        
#> [13] utf8_1.1.4       cli_1.0.1        htmltools_0.3.6  yaml_2.2.0      
#> [17] digest_0.6.18    assertthat_0.2.0 crayon_1.3.4     purrr_0.3.0     
#> [21] glue_1.3.0       evaluate_0.12    rmarkdown_1.11   stringi_1.2.4   
#> [25] compiler_3.5.2   pillar_1.3.1     pkgconfig_2.0.2
```

<sup>Created on 2019-02-17 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
</details>

**Comment by hadley** [comment]
Would you mind first making a simpler example?

**Comment by gtm19** [comment]
> Would you mind first making a simpler example?

No problem @hadley. All done.

**Comment by hadley** [comment]
Thanks, that's perfect!

**Comment by hadley** [comment]
@AmeliaMN do you want to have a look at this?

**Comment by gtm19** [comment]
@AmeliaMN I have added some further comments on the [original pull request](https://github.com/tidyverse/forcats/pull/173) which might help also.

I have another pull request - with tests to highlight this issue added - ready to go, but didn't want to spam you with pull requests until you (and @hadley) had agreed with the sentiment in my spam comments 😜

**Comment by gtm19** [comment]
@batpigandme regarding the [`fct_collapse` issue](https://github.com/tidyverse/forcats/issues/172), you may want to check out pull request (#176) which fixes this issue, as well as reordering the factor levels so "other" is last, for consistency with other functions in the package.

**Comment by avishaitsur** [comment]
The reprex still results in incorrect levels.
Is there a solution?

**Comment by gtm19** [comment]
I don't believe this was fixed in 0.4.0, but my pull request #176 contains a solution.
